### ISSUE TITLE
Font Scaling Problem (Hi-res Font Support)

### ISSUE THREAD

**Issue by brettg94** [LazyWizard/lazylib#5]
I found a small oversight in `LazyFont`'s `FontLoader.loadFont(...)` method.

Current code on branch `master` lines(a quick check on `next` shows the same code too):
```
// Load the font image into a texture
// TODO: Add support for multiple image files; 'pages' in the font file
// (this is a low priority as no vanilla font uses multiple pages)
val textureId: Int
val textureWidth: Float
val textureHeight: Float
try {
  Global.getSettings().loadTexture(imgFile)
  val texture = Global.getSettings().getSprite(imgFile)
  textureId = texture.textureId
->  textureWidth = texture.width
->  textureHeight = texture.height
} catch (ex: IOException) {
  throw FontException("Failed to load texture atlas '$imgFile'", ex)
}
```
`line 97-111`

The width and height are being read from the atlas's texture file rather than the .fnt file metadata.
This is fine and works without issue for all standard Starsector fonts. However, if the resolution of the font atlas texture differs from the specified texture scale in the metadata (`scaleW=xxx scaleH=xxx`), the font's characters aren't mapped to the correct coordinates, resulting in a garbled mess of pixels in areas where legible text should be rendered.

`LazyFont`'s implementation differs from how Starsector's internal font system handles this situation, which appears to scale the metadata's coordinates based on the size of the texture atlas.

To elaborate on why this is an issue for me:
I play with resolution scaling on, which results in blurry fonts by default. I upscaled all of the game's font files by a factor of 2 (eg 128x128 => 256x256). This results in more legible, crisp text for the base game.

However, modded content that leverages LazyLib's font system (and relies upon the base game's font files) results in illegible text due to the different behavior between these two systems.

I propose the following fix:
```
...
try {
  Global.getSettings().loadTexture(imgFile)
  val texture = Global.getSettings().getSprite(imgFile)
  textureId = texture.textureId
} catch (ex: IOException) {
  throw FontException("Failed to load texture atlas '$imgFile'", ex)
}
val scaleW = java.lang.Float.parseFloat(metadata[31])
val scaleH = java.lang.Float.parseFloat(metadata[33])
val font = LazyFont(fontName, textureId, baseHeight, scaleW, scaleH)
...
```

This change results in legible text while retaining the newly added hi-res textures. Additionally, there appears to be no problems when using the game's standard normal-res fonts with this change in place.

![lazyfont-comparison](https://github.com/user-attachments/assets/26e66d08-5477-4189-8d90-8aea20750965)

Let me know you'd like me to open a PR for this, or if there's an argument to be made for keeping the current behavior. Thanks!

**Comment by LazyWizard** [comment]
Thanks for the report! It was my bad using the actual texture width/height; I'd assumed scaleW and scaleH were only there to make life more convenient for file parsers and would always be identical to the actual texture dimensions.

You can open a pull request if you'd like, or I can make the change myself. Either way, I'll credit you in the changelog. What name would you want the credit under?

**Comment by brettg94** [comment]
No worries man, I was under a similar assumption until I dumped the hi-res textures in and it just worked.

PR: #6 

I opened it against `next`, hopefully that's okay. You can credit me as `psiyon` if you really want.

Thanks again!

**Comment by LazyWizard** [comment]
Merged #6, marking as closed. Thank you again!

And are you [this Psiyon?](https://www.youtube.com/@Psiyon/videos) If so, it's good to see you around!
